### ISSUE TITLE
New Rule: no-unpublished-bin

### ISSUE THREAD

**Issue by mysticatea** [mysticatea/eslint-plugin-node#49]
This rule will warn the whole file which is unpublished and which exists in `bin` field of `package.json`.

Example:

**package.json**

```json
{
    "bin": "bin/foo.js",
    "files": ["lib"]
}
```

In this case, `bin/foo.js` is warned.
`bin/foo.js` is unpublished but is specified as `bin` scripts.
This is going to cause installation failure (failed to make symlink for the command).

**Comment by mysticatea** [comment]
Fixed in afa751a7f39b7453d238496a261d4b776a7b2585
